diff -rNu gobject-introspection-1.75.6/meson.build gobject-introspection-1.75.6.new/meson.build
--- gobject-introspection-1.75.6/meson.build	2023-02-14 01:59:24.000000000 +0200
+++ gobject-introspection-1.75.6.new/meson.build	2023-04-18 11:01:02.972025907 +0300
@@ -161,6 +161,8 @@
   g_ir_scanner_env.append('PKG_CONFIG_PATH', meson.global_build_root() + '/meson-uninstalled')
   g_ir_scanner_env.set('PKG_CONFIG', pkg_config.full_path())
 endif
+host_cc = meson.get_compiler('c', native : false).cmd_array()[0]
+g_ir_scanner_env.append('CC',host_cc)
 
 configure_file(
   configuration: config,
