diff -rNu gstreamer-1.20.7/gst/meson.build gstreamer-1.20.7.new/gst/meson.build
--- gstreamer-1.20.7/gst/meson.build	2023-07-26 15:18:38.000000000 +0000
+++ gstreamer-1.20.7.new/gst/meson.build	2025-12-06 07:45:05.972244544 +0000
@@ -273,6 +273,18 @@
       '--cflags-end']
   endif
 
+  gst_gir_extra_args += ['--lib-dirs-envvar=DUMMY_UNUSED_ENVVAR',
+    '--use-binary-wrapper=' + get_option('binary_wrapper'), '--use-ldd-wrapper=' + get_option('ldd_wrapper')]
+
+  gcc = meson.get_compiler('c', native : false).cmd_array()[0]
+  cpp = gcc.replace('gcc','cpp')
+  cflags_list = get_option('c_args')
+  cflags = ''
+  foreach flag : cflags_list
+    cflags = cflags + flag + ' '
+  endforeach
+  sysroot = meson.get_external_property('root')
+
   gst_incdirs += [configinc]
   gir = {
     'sources' : gst_sources + gst_headers + gst_enums + [gst_version_h],
@@ -283,7 +295,9 @@
     'export_packages' : pkg_name,
     'includes' : ['GLib-2.0', 'GObject-2.0', 'GModule-2.0' ],
     'install' : true,
+    'include_directories' : [ sysroot + '/share/gir-1.0' ],
     'extra_args' : gst_gir_extra_args,
+    'env' : {'CPP' : cpp, 'CFLAGS' : cflags}
   }
   library_def += {'gir': [gir]}
   # When building statically the information about the gir is going to be
diff -rNu gstreamer-1.20.7/libs/gst/base/meson.build gstreamer-1.20.7.new/libs/gst/base/meson.build
--- gstreamer-1.20.7/libs/gst/base/meson.build	2023-07-26 15:18:38.000000000 +0000
+++ gstreamer-1.20.7.new/libs/gst/base/meson.build	2025-12-06 07:47:51.062776336 +0000
@@ -62,6 +62,15 @@
 
 if build_gir
   gst_gir_extra_args = gir_init_section + [ '--c-include=gst/base/base.h' ]
+  gst_gir_extra_args += ['--lib-dirs-envvar=DUMMY_UNUSED_ENVVAR',
+    '--use-binary-wrapper=' + get_option('binary_wrapper'), '--use-ldd-wrapper=' + get_option('ldd_wrapper')]
+  gcc = meson.get_compiler('c', native : false).cmd_array()[0]
+  cpp = gcc.replace('gcc','cpp')
+  cflags_list = get_option('c_args')
+  cflags = ''
+  foreach flag : cflags_list
+    cflags = cflags + flag + ' '
+  endforeach
   gir = {
     'sources' : gst_base_sources + gst_base_headers,
     'namespace' : 'GstBase',
@@ -74,6 +83,7 @@
     'includes' : ['GLib-2.0', 'GObject-2.0', 'GModule-2.0', 'Gst-1.0'],
     'install' : true,
     'extra_args' : gst_gir_extra_args,
+    'env' : {'CPP' : cpp, 'CFLAGS' : cflags}
   }
 
   library_def += {'gir': [gir]}
diff -rNu gstreamer-1.20.7/libs/gst/check/meson.build gstreamer-1.20.7.new/libs/gst/check/meson.build
--- gstreamer-1.20.7/libs/gst/check/meson.build	2023-07-26 15:18:38.000000000 +0000
+++ gstreamer-1.20.7.new/libs/gst/check/meson.build	2025-12-06 07:49:23.061627143 +0000
@@ -65,6 +65,8 @@
 gst_check_gen_sources = []
 if build_gir
   gst_gir_extra_args = gir_init_section + [ '--c-include=gst/check/check.h' ]
+  gst_gir_extra_args += ['--lib-dirs-envvar=DUMMY_UNUSED_ENVVAR',
+    '--use-binary-wrapper=' + get_option('binary_wrapper')]
   gir = {
     'sources' : gst_check_sources + gst_check_headers,
     'namespace' : 'GstCheck',
diff -rNu gstreamer-1.20.7/libs/gst/controller/meson.build gstreamer-1.20.7.new/libs/gst/controller/meson.build
--- gstreamer-1.20.7/libs/gst/controller/meson.build	2023-07-26 15:18:38.000000000 +0000
+++ gstreamer-1.20.7.new/libs/gst/controller/meson.build	2025-12-06 07:52:02.726116860 +0000
@@ -60,6 +60,15 @@
 
 if build_gir
   gst_gir_extra_args = gir_init_section + [ '--c-include=gst/controller/controller.h' ]
+  gst_gir_extra_args += ['--lib-dirs-envvar=DUMMY_UNUSED_ENVVAR',
+    '--use-binary-wrapper=' + get_option('binary_wrapper'), '--use-ldd-wrapper=' + get_option('ldd_wrapper')]
+  gcc = meson.get_compiler('c', native : false).cmd_array()[0]
+  cpp = gcc.replace('gcc','cpp')
+  cflags_list = get_option('c_args')
+  cflags = ''
+  foreach flag : cflags_list
+    cflags = cflags + flag + ' '
+  endforeach
   gir = {
     'sources' : gst_controller_sources + gst_controller_headers + [gstcontroller_h] + [gstcontroller_c],
     'namespace': 'GstController',
@@ -72,6 +81,7 @@
     'includes' : ['GLib-2.0', 'GObject-2.0', 'GModule-2.0', 'Gst-1.0'],
     'install' : true,
     'extra_args' : gst_gir_extra_args,
+    'env' : {'CPP' : cpp, 'CFLAGS' : cflags}
   }
   library_def += {'gir': [gir]}
   if not static_build
diff -rNu gstreamer-1.20.7/libs/gst/net/meson.build gstreamer-1.20.7.new/libs/gst/net/meson.build
--- gstreamer-1.20.7/libs/gst/net/meson.build	2023-07-26 15:18:38.000000000 +0000
+++ gstreamer-1.20.7.new/libs/gst/net/meson.build	2025-12-06 07:53:19.932838747 +0000
@@ -47,6 +47,15 @@
 
 if build_gir
   gst_gir_extra_args = gir_init_section + [ '--c-include=gst/net/net.h' ]
+  gst_gir_extra_args += ['--lib-dirs-envvar=DUMMY_UNUSED_ENVVAR',
+    '--use-binary-wrapper=' + get_option('binary_wrapper'), '--use-ldd-wrapper=' + get_option('ldd_wrapper')]
+  gcc = meson.get_compiler('c', native : false).cmd_array()[0]
+  cpp = gcc.replace('gcc','cpp')
+  cflags_list = get_option('c_args')
+  cflags = ''
+  foreach flag : cflags_list
+    cflags = cflags + flag + ' '
+  endforeach
   gir = {
     'sources' : gst_net_sources + gst_net_headers,
     'namespace' : 'GstNet',
@@ -59,6 +68,7 @@
     'includes' : ['GLib-2.0', 'GObject-2.0', 'GModule-2.0', 'Gio-2.0', 'Gst-1.0'],
     'install' : true,
     'extra_args' : gst_gir_extra_args,
+    'env' : {'CPP' : cpp, 'CFLAGS' : cflags}
   }
   library_def += {'gir': [gir]}
   if not static_build
diff -rNu gstreamer-1.20.7/meson_options.txt gstreamer-1.20.7.new/meson_options.txt
--- gstreamer-1.20.7/meson_options.txt	2023-07-26 15:18:38.000000000 +0000
+++ gstreamer-1.20.7.new/meson_options.txt	2025-12-06 07:55:12.140882376 +0000
@@ -46,3 +46,10 @@
        description : 'package origin URL to use in plugins')
 option('doc', type : 'feature', value : 'auto', yield: true,
        description: 'Enable documentation.')
+       
+option('binary_wrapper', type: 'string',
+  description: 'Use a wrapper to run gicompiler and binaries produced by giscanner (useful when cross-compiling)'
+)
+option('ldd_wrapper', type: 'string',
+  description: 'Use a wrapper to run ldd used by giscanner (useful when cross-compiling)'
+)
